src/controller/SetUpCodePairer.cpp

```diff
@@ -497,6 +498,17 @@ bool SetUpCodePairer::NodeMatchesCurrentFilter(const Dnssd::DiscoveredNodeData &
         return false;
     }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
```

Comment:
Shouldn't this be living in `src/controller/jcm/AutoCommissioner.cpp` (with a virtual hook as needed; either make NodeMatchesCurrentFilter virtual, or have a virtual check for "has valid commissioning mode", which will be implemented as "!= 0" in the base class and the subclass can do whatever it wants)?